src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    static final int FIRST_SPACE_INDEX = 10;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Report for period " + dateFrom + " - " + dateTo);
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate endDate = LocalDate.parse(dateTo, formatter);
+        for (String name : names) {
+            stringBuilder.append(System.lineSeparator());
+            stringBuilder.append(calculateSalaryForEmployee(data, name, startDate, endDate));
+        }
+        return stringBuilder.toString();
+    }
+
+    private String calculateSalaryForEmployee(String[] data, String name,
+                                              LocalDate startDate, LocalDate endDate) {
+        int salary = 0;
+        for (String dataLine : data) {
+            if (dataLine.contains(name)) {
+                salary += calculateIfApplicable(dataLine, startDate, endDate);
+            }
+        }
+        return name + " - " + salary;
+    }
+
+    private int calculateIfApplicable(String dataLine, LocalDate startDate, LocalDate endDate) {
```

Comment:
```suggestion
    private int getSalary(String[] line, LocalDate startDate, LocalDate endDate) {
```
let's pass here splitted line